ddtrace/contrib/aiokafka/patch.py

```diff
@@ -0,0 +1,175 @@
+from ddtrace.internal.schema import schematize_service_name
+from ddtrace.internal.schema.span_attribute_schema import SpanDirection
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.pin import Pin
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+
+config._add(
+    "aiokafka",
+    dict(
+        _default_service=schematize_service_name("kafka"),
+        distributed_tracing_enabled=asbool(os.getenv("DD_KAFKA_PROPAGATION_ENABLED", default=False)),
+    ),
+)
+
+
+def get_version() -> str:
+    return getattr(aiokafka, "__version__", "")
+
+
+def patch():
+    if getattr(aiokafka, "_datadog_patch", False):
+        return
+    aiokafka._datadog_patch = True
+
+    _w("aiokafka", "AIOKafkaProducer.send_and_wait", traced_send_and_wait)
+    _w("aiokafka", "AIOKafkaConsumer.getone", traced_getone)
+
+    Pin().onto(aiokafka.AIOKafkaProducer)
+    Pin().onto(aiokafka.AIOKafkaConsumer)
+
+
+def unpatch():
+    if getattr(aiokafka, "_datadog_patch", False):
+        return
+
+    aiokafka._datadog_patch = False
+
+    _u(aiokafka.AIOKafkaProducer, "send_and_wait")
+    _u(aiokafka.AIOKafkaConsumer, "getone")
+
+
+def bootstrap_servers(instance):
+    if hasattr(instance, "client"):
+        client = instance.client
+    if hasattr(instance, "_client"):
+        client = instance._client
+    if client._bootstrap_servers is not None:
+        return ",".join(client._bootstrap_servers)
+
+
+def create_send_span_tags(instance, args, kwargs):
+    topic = get_argument_value(args, kwargs, 0, "topic")
+    tags = {
+        COMPONENT: config.aiokafka.integration_name,
+        SPAN_KIND: SpanKind.PRODUCER,
+        TOPIC: topic,
+        HOST_LIST: bootstrap_servers(instance),
+        MESSAGING_SYSTEM: SERVICE,
+    }
+    partition = get_argument_value(args, kwargs, 3, "partition", True) or None
+    if partition:
+        tags[PARTITION] = partition
+    key = get_argument_value(args, kwargs, 2, "key", True) or None
+    if key:
+        tags[MESSAGE_KEY] = key.decode("utf-8")
+    value = get_argument_value(args, kwargs, 1, "value", True) or None
+    tags[TOMBSTONE] = str(value is None).lower()
+    return tags
+
+
+async def traced_send_and_wait(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    with core.context_with_data(
+        "aiokafka.send_and_wait",
+        parent=None,
+        span_name=schematize_messaging_operation(PRODUCE, provider="kafka", direction=SpanDirection.OUTBOUND),
+        span_type=SpanTypes.WORKER,
+        service=trace_utils.ext_service(pin, config.aiokafka),
+        tags=create_send_span_tags(instance, args, kwargs),
+        pin=pin,
+    ) as ctx:
+        result = await func(*args, **kwargs)
+        core.dispatch("aiokafka.send_and_wait.post", (ctx,))
+        return result
```

Comment:
This pattern closes the span when the `with` block exits, without the need for a `dispatch` call.

```suggestion
    ) as ctx, ctx[ctx["call_key"]]:
        result = await func(*args, **kwargs)
        return result
```